src/Domain/DonationService.php

```diff
@@ -323,4 +327,67 @@ private function runWithPossibleRetry(\Closure $retryable, string $actionName):
             }
         }
     }
+
+    /**
+     * Finalized a donation, instructing stripe to attempt to take payment.
+     */
+    public function confirm(
+        Donation $donation,
+        string $paymentMethodId
+    ): \Stripe\PaymentIntent {
+        $this->updateDonationFees($paymentMethodId, $donation);
+
+        // looks like sometimes $paymentIntentId and $paymentMethodId are for different customers.
```

Comment:
I know this comment is migrated from a Confirm Action edit a few months ago, but I'm struggling to understand exactly why it would be the case or how it relates to the next line. Do we still need it, and if so can we clarify?

Reply:
good question, I'm not sure. Let me try and find some context from when it was originally written.

Reply:
So this is from issue MAT-339 which we never quite got to the bottom of. Seems to have been some isolated cases and we don't know if they were related to genuine donation attempts. Shall we just delete the comment?

Reply:
Yeah I think so. We're not knowingly _supporting_ cases where different Customers own the two objects, right?

Reply:
We're not. I've deleted it.